### PR TITLE
Update PR template

### PR DESCRIPTION
**Pull request by JanuszL**
- changes the PR template in the way to make it clear what can be removed
  and what now, and what kind of answers are expected

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
- It improves PR template used in the project

#### What happened in this PR?
 - changes the PR template in the way to make it clear what can be removed and what now, and what kind of answers are expected
 - no docs are affected
 - no testing is necessary

**JIRA TASK**: [NA]

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
 #### What happened in this PR?
- - Explain solution of the problem, new feature added.
- - What was changed, added, removed?
- - What is most important part that reviewers should focus on?
- - Was this PR tested? How?
- - Were docs and examples updated, if necessary?
+*Fill relevant points, put NA otherwise*
+ - Problem solution in this PR:
+ 
+   *[ Explain solution of the problem, new feature added here. ]*
```
Comment (mzient) [inline .github/PULL_REQUEST_TEMPLATE.md]: Strange formatting - shouldn't this line stick to the previous?

**Comment by klecki** [comment]
I'm not sure if adding such extensive blocks of text to edit out/remove when filling in the template is convenient.

For example I did the
```
**JIRA TASK**: [DALI-XXXX]
``` 
in such way that you can easily double click on the number and replace it. Now it's several word to edit out.

Also what about the markdown, and what parts should the user leave?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (JanuszL) [reply .github/PULL_REQUEST_TEMPLATE.md]: Fixed

**Comment by JanuszL** [comment]
> I'm not sure if adding such extensive blocks of text to edit out/remove when filling in the template is convenient.
> 
> For example I did the
> 
> ```
> **JIRA TASK**: [DALI-XXXX]
> ```
> 
> in such way that you can easily double click on the number and replace it. Now it's several word to edit out.
> 
> Also what about the markdown, and what parts should the user leave?

I don't have any particular feeling about `[DALI-XXXX]`, can leave it as it was. I have also added `*Fill relevant points, put NA otherwise. Replace anyting inside []*` to make it more clear.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
 #### What happened in this PR?
- - Explain solution of the problem, new feature added.
- - What was changed, added, removed?
- - What is most important part that reviewers should focus on?
- - Was this PR tested? How?
- - Were docs and examples updated, if necessary?
+*Fill relevant points, put NA otherwise. Replace anything inside []*
+ - Problem solution in this PR:
+     *[ Explain solution of the problem, a new feature added here. ]*
+ - Affected files and modules:
```
Comment (mzient) [inline .github/PULL_REQUEST_TEMPLATE.md]: I think the previous point was better - this one looks like asking for a list of files, but PR mechanism automates this.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
 - Refactoring to improve *what*
 
 #### What happened in this PR?
- - Explain solution of the problem, new feature added.
- - What was changed, added, removed?
- - What is most important part that reviewers should focus on?
- - Was this PR tested? How?
- - Were docs and examples updated, if necessary?
+*Fill relevant points, put NA otherwise. Replace anything inside []*
+ - Problem solution in this PR:
```
Comment (mzient) [inline .github/PULL_REQUEST_TEMPLATE.md]: Maybe a bit Grammar Nazi on my side, but the title "Problem solution in this PR" sounds like "state the problem which was solved" - not "what solution was applied".

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
+     *[ Explain solution of the problem, a new feature added here. ]*
+ - Affected files and modules:
+     *[ Describe here what was changed, added, removed. ]*
+ - Key points relevant for the review:
```
Comment (mzient) [inline .github/PULL_REQUEST_TEMPLATE.md]: :+1:

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (JanuszL) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (JanuszL) [reply .github/PULL_REQUEST_TEMPLATE.md]: Adjusted a bit

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,14 +1,21 @@
...
 #### What happened in this PR?
- - Explain solution of the problem, new feature added.
- - What was changed, added, removed?
- - What is most important part that reviewers should focus on?
- - Was this PR tested? How?
- - Were docs and examples updated, if necessary?
+*Fill relevant points, put NA otherwise. Replace anything inside []*
+ - What solution was applied:
+     *[ Explain solution of the problem, a new feature added here. ]*
+ - Affected modules and functionalities:s
```
Comment (klecki) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
 - Affected modules and functionalities:
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,14 +1,21 @@
...
+     *[ Describe here what is the most important part that reviewers should focus on. ]*
+ - Validation and testing:
+     *[ Describe here if and how this PR is tested. ]*
+ - Documentation (including examples):
+     *[ Describe here if documentation and examples were updated. ]*
 
-**JIRA TASK**: [DALI-XXXX]
\ No newline at end of file
+
+**JIRA TASK**: [*Use DALI-XXXX or NA*]
```
Comment (klecki) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
**JIRA TASK**: *[Use DALI-XXXX or NA]*
```

**Comment by JanuszL** [comment]
> I don't heave any strong opinion outside of the fact that this requires a little bit more of editing the PR message.

That was the point. That you need to think before you write and make it meaningful.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (JanuszL) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (JanuszL) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done

**Pull request merged**